launcher/src/main/java/org/apache/spark/launcher/package-info.java

```diff
@@ -49,6 +49,39 @@
  * </pre>
  *
  * <p>
+ * Currently, while launching spark application with
+ * {@link org.apache.spark.launcher.SparkLauncher#startApplication(
+ * org.apache.spark.launcher.SparkAppHandle.Listener...)}, there are two options available
+ * for YARN manager in cluster deploy mode:
+ *  - to launch Spark Application as a Thread inside current JVM using
+ *    the {@link org.apache.spark.launcher.SparkLauncher#launchAsThread(boolean)}
+ *  - to request application be killed if launcher process exits using
+ *    the {@link org.apache.spark.launcher.SparkLauncher#stopIfLauncherShutdown()}.
```

Comment:
sorry should have mentioned this first time but I think we should reorder things a bit.  Perhaps move the details of being able to launch as a thread or as a process up to the general description at line 23.  just stating the launcher support both of these modes and add restrictions for thread mode (yarn cluster only). 

reword the description of this just to be more like an example, here is launching in thread mode for yarn cluster mode with the stop if launcher shutdown option enabled.